### ISSUE TITLE
Some reduction operators have double signatures

### ISSUE THREAD

**Issue by heitorschueroff** [pytorch/pytorch#61486]
Some reduction operators have two signatures, one that takes a `dim` and `keepdim` parameter to reduce along one or more dimensions and another that does not which reduces the flattened tensor. We should update these operators to only have one signature with an optional `dim=None` parameter which is more compatible with the Python Array API Standard.

## Operators with double signature to update

- [ ] [argmax](https://pytorch.org/docs/master/generated/torch.argmax.html#torch.argmax)
- [ ] [all](https://pytorch.org/docs/master/generated/torch.all.html#torch.all)
- [ ] [any](https://pytorch.org/docs/master/generated/torch.any.html#torch.any)
- [ ] [max](https://pytorch.org/docs/master/generated/torch.max.html#torch.max)
- [ ] [min](https://pytorch.org/docs/master/generated/torch.min.html#torch.min)
- [ ] [mean](https://pytorch.org/docs/master/generated/torch.mean.html#torch.mean)
- [ ] [median](https://pytorch.org/docs/master/generated/torch.median.html#torch.median)
- [ ] [nanmedian](https://pytorch.org/docs/master/generated/torch.nanmedian.html#torch.nanmedian)
- [ ] [nansum](https://pytorch.org/docs/master/generated/torch.nansum.html#torch.nansum)
- [ ] [prod](https://pytorch.org/docs/master/generated/torch.prod.html#torch.prod)
- [ ] [std](https://pytorch.org/docs/master/generated/torch.std.html#torch.std)
- [ ] [std_mean](https://pytorch.org/docs/master/generated/torch.std_mean.html#torch.std_mean)
- [ ] [sum](https://pytorch.org/docs/master/generated/torch.sum.html#torch.sum)
- [ ] [var](https://pytorch.org/docs/master/generated/torch.var.html#torch.var)
- [ ] [var_mean](https://pytorch.org/docs/master/generated/torch.var_mean.html#torch.var_mean)
- [ ] [count_nonzero](https://pytorch.org/docs/master/generated/torch.count_nonzero.html#torch.count_nonzero) [missing keepdim]

**Comment by rgommers** [comment]
I think this is still blocked, first gh-29137 needs to be resolved for handling the `dim=None`, `dim=()` cases the same as `axis=` in NumPy and the array API standard?

**Comment by heitorschueroff** [comment]
@rgommers We should fix the `dim=None` and `dim=()` behavior but I don't think that needs to block this issue necessarily. I see them as two separate issues that both need to be resolved, or am I missing something?

**Comment by rgommers** [comment]
Maybe I'm just misremembering, I thought having a signature with an optional `dim=None` was not possible until gh-29137 was resolved (that's what my last comment on that issue says at least). But maybe that's no longer true, and it's possible now?

**Comment by heitorschueroff** [comment]
@rgommers I see what you mean, this is blocked on this issue https://github.com/pytorch/pytorch/issues/44409
